fuzz/fuzz_targets/simple_model.rs

```diff
@@ -5,98 +5,59 @@
-	let dir = tempdir().unwrap();
-	let options = parity_db::Options {
-		path: dir.path().to_owned(),
-		columns: vec![parity_db::ColumnOptions {
-			compression: match config.compression {
-				CompressionType::NoCompression => parity_db::CompressionType::NoCompression,
-				CompressionType::Snappy => parity_db::CompressionType::Snappy,
-				CompressionType::Lz4 => parity_db::CompressionType::Lz4,
-			},
-			btree_index: config.btree_index,
-			..parity_db::ColumnOptions::default()
-		}],
-		sync_wal: true,
-		sync_data: true,
-		stats: false,
-		salt: None,
-	};
-	let mut db = parity_db::Db::open_or_create(&options).unwrap();
-	let mut model = BTreeMap::<u8, u8>::default();
-	for action in actions {
-		// We apply the action on both the database and the model
-		match action {
-			Action::Transaction(operations) => {
-				db.commit(
-					operations.iter().copied().map(|(k, v)| (0u8, vec![k], v.map(|v| vec![v]))),
-				)
-				.unwrap();
-				for (k, v) in operations {
-					if let Some(v) = v {
-						model.insert(k, v);
-					} else {
-						model.remove(&k);
-					}
-				}
-			},
-			Action::Restart =>
-				db = {
-					drop(db);
-					parity_db::Db::open_or_create(&options).unwrap()
-				},
+use parity_db_fuzz::*;
+use std::{
+	collections::{BTreeMap, HashMap},
+	path::Path,
+};
+
+struct Simulator;
+
+impl DbSimulator for Simulator {
+	type Operation = (u8, Option<u8>);
+	type Model = BTreeMap<u8, u8>;
+
+	fn build_options(config: &Config, path: &Path) -> parity_db::Options {
+		parity_db::Options {
+			path: path.to_owned(),
+			columns: vec![parity_db::ColumnOptions {
+				compression: config.compression.into(),
+				btree_index: config.btree_index,
+				..parity_db::ColumnOptions::default()
+			}],
+			sync_wal: true,
+			sync_data: true,
+			stats: false,
+			salt: None,
+			compression_threshold: HashMap::new(),
 		}
+	}
 
-		// We check the state
-		for (k, v) in &model {
-			assert_eq!(db.get(0, &[*k]).unwrap(), Some(vec![*v]));
+	fn apply_operation_on_model(operation: &(u8, Option<u8>), model: &mut Self::Model) {
+		let (k, v) = operation;
+		if let Some(v) = *v {
+			model.insert(*k, v);
+		} else {
+			model.remove(k);
 		}
+	}
 
-		if config.btree_index {
-			// We check the BTree forward iterator
-			let mut model_content = model.iter().map(|(k, v)| (*k, *v)).collect::<Vec<_>>();
-			let mut db_iter = db.iter(0).unwrap();
-			db_iter.seek_to_first().unwrap();
-			let mut db_content = Vec::new();
-			while let Some((k, v)) = db_iter.next().unwrap() {
-				db_content.push((k[0], v[0]));
-			}
-			assert_eq!(db_content, model_content);
-
-			// We check the BTree backward iterator
-			model_content.reverse();
-			db_iter.seek_to_last().unwrap();
-			let mut db_content = Vec::new();
-			while let Some((k, v)) = db_iter.prev().unwrap() {
-				db_content.push((k[0], v[0]));
-			}
-			assert_eq!(db_content, model_content);
+	fn map_operation(operation: &(u8, Option<u8>)) -> parity_db::Operation<Vec<u8>, Vec<u8>> {
```

Comment:
Could have an util function for sharing this logic with ref counted (for now it's not very useful actually:)